required/latex-lab/latex-lab-footnotes.dtx

```diff
@@ -0,0 +1,1731 @@
+         \LogHook{fntext/begin}
+         \LogHook{fntext/end}
+         \LogHook{fntext/after}
+%    \end{macrocode}
+%    Show the info only once (if at all).
+%    \begin{macrocode}
+         \cs_gset_eq:NN \__fnote_debug_footnotetext: \prg_do_nothing:
+       }
+  }
+
+\cs_new:Npn \fnote_footnotetext:n #1 {
+  \__fnote_debug_footnotetext:
+%-------
+% ./linguex/linguex.sty
+  \UseHook{fntext/before}
+%-------
+  \@kernel@process@footnotetext
+  \@footnotetext@cfgpoint {  % config point
+%-------
+% resetting baselinestretch ... (could be done further down)
+% ./uafthesis/uafthesis.cls
+% ./setspace/setspace.sty
+% ./footmisc/footmisc.sty (normal)
+    \UseHook{fntext}
+%-------
+    \reset@font
+    \footnotesize
+%-------
+% some classes use a different font size, e.g.,
+% ./nrc/nrc1.cls  ./nrc/nrc2.cls
+% but those could be done in fntext/para instead
+%-------
+%    \end{macrocode}
+%    In case of sidenotes the next settings are pointless, but as they
+%    do not hurt (except for the \cs{hsize} setting) and are needed
+%    for all other cases we make them here and overwrite them for side notes
+%    \begin{macrocode}
+    \interlinepenalty\interfootnotelinepenalty
+    \splittopskip\footnotesep
+    \splitmaxdepth \dp\strutbox
+    \floatingpenalty \@MM
+    \hsize\columnwidth
+    \@parboxrestore
+    \parindent 1em     % typical default used in \@makefntext moved up here
+    \def\@currentcounter{footnote}
+    \protected@edef \@currentlabel { \p@footnote \@thefnmark }
+%-------
+% for altering para parameters ...
+% code for resphilosophica came earlier but it could go here.
+% Has the advantage that one can also overwrite \cs{@currentcounter}
+% and \cs{@currentlabel} is that is necessary.
+%
+% ./resphilosophica/resphilosophica.cls
+    \UseHook{fntext/para}
+%-------
+    \color@begingroup
+%-------
+% fnpara wants to replace \@makefntext{...} and para and side option of footmisc etc too ...
+% so we make this a config point
+%-------
+      \@footnotetext@cfgpointii       % config point
+        {
+%-------
+% ./resphilosophica/resphilosophica.cls
+%-------
+          \@footnotetext@cfgpointiii  % config point
+%-------
+% bibarts
+% fnbreak.sty
+          \UseHook{fntext/begin}
+%-------
+          \ignorespaces
+          #1
+%-------
+% bibarts
+% fnbreak.sty
+          \UseHook{fntext/end}
+%-------
+          \@footnotetext@cfgpointiv   % config point
+        }
+      \par
+    \color@endgroup
+  }
+%-------
+% ./linguex/linguex.sty
+  \UseHook{fntext/after}
+%-------
+}
+
+% default for config point (1 arg)
+\def\@footnotetext@cfgpoint    { \insert\footins }
+
+% default for config point (1 arg)
+\def\@footnotetext@cfgpointii  { \@makefntext }
+
+
+% default for config point (0 args)
+\def\@footnotetext@cfgpointiii { \rule\z@\footnotesep }
+
+% default for config point (0 args)
+\def\@footnotetext@cfgpointiv  { \@finalstrut\strutbox }
+
+
+% kernel hook for tagging (2 args)
+\cs_new:Npn \@kernel@process@footnotetext {}
```

Comment:
You are not exactly consistent in what to use: could we have `\cs_new_(protected):Npn` for all the new material? I fully understand sticking to LaTeX2e names for code you are retaining from existing implementation.